src/main/java/net/mcreator/minecraft/api/ModAPI.java

```diff
@@ -38,21 +38,7 @@ void setImplementations(Map<String, Implementation> implementations) {
 		this.implementations = implementations;
 	}
 
-	public static class Implementation {
-
-		public final String gradle;
-		@Nullable public final List<String> update_files;
-		public final ModAPI parent;
-		public final boolean requiredWhenEnabled;
-
-		public Implementation(ModAPI parent, String gradle, @Nullable List<String> update_files,
-				boolean requiredWhenEnabled) {
-			this.gradle = gradle;
-			this.update_files = update_files;
-			this.parent = parent;
-			this.requiredWhenEnabled = requiredWhenEnabled;
-		}
-
-	}
+	public record Implementation(ModAPI parent, String gradle, @Nullable List<String> update_files,

```

Comment:
Did you test this, according to https://github.com/google/gson/issues/1794, records are not working with GSON

Reply:
>records are not working with GSON

Because their parameters are final, damn...

Reply:
You may need to recheck all changes to see which ones are used by GSON anywhere

Reply:
Is API info stored in **${modid}.mcreator** files?

Reply:
It is stored in plugin, not 100% sure if gson parses it, though, check in the code to be sure

Reply:
Plugins store them in YAML, it's OKay